azure-iot-device/azure/iot/device/common/alarm.py

```diff
@@ -0,0 +1,36 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+from threading import Thread, Event
+import time
+
+
+class Alarm(Thread):
+    """Call a function at a specified time"""
+
+    def __init__(self, alarm_time, function, args=None, kwargs=None):
+        Thread.__init__(self)
+        self.alarm_time = alarm_time
+        self.function = function
+        self.args = args if args is not None else []
+        self.kwargs = kwargs if kwargs is not None else {}
+        self.finished = Event()
+
+    def cancel(self):
+        """Stop the alarm if it hasn't finished yet."""
+        self.finished.set()
```

Comment:
>self.finished.set() [](start = 8, length = 19)

Do you want to null out function, args, and kwargs since the function will never be called at this point?  GC _should_ clean them up, but I don't necessarily trust GC.
(note: be careful if you do this -- the easy answer has a race condition.)

Reply:
This pattern is copied from the other similar classes (e.g. Timer) from the builtin threading module. I'm inclined to keep the similarities since the builtins are known to be reliable.